### PR TITLE
add unocss customization

### PR DESCRIPTION
**Pull request by omar2205**
With this PR, you can customize UnoCSS with presets and all the fun stuff.

Example, to change the font
```ts
// ./unocss.ts
import presetWebFonts from 'https://esm.sh/@unocss/preset-web-fonts@0.42.0'
import presetUno from 'https://esm.sh/@unocss/preset-uno@0.42.0'

export const UNOCSS_OPTS = {
  presets: [
    presetUno(),
    presetWebFonts({
      provider: 'google', // default provider
      fonts: {
        // these will extend the default theme
        sans: {
          name: 'Outfit',
          weights: ['100', '300', '400', '700'],
        },
        mono: ['Fira Code', 'Fira Mono:400,700'],
      },
    }),
  ],
  preflights: [
    {
      getCSS: ({ theme }) => `
        html {
          font-family: 'Outfit';
        }
      `
    }
  ]
}
```

```ts
// main.ts
// ...
import { UNOCSS_OPTS } from './unocss.ts'

blog({
  // ...
  unocss: UNOCSS_OPTS
})
```

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export interface BlogSettings {
   lang?: string;
   timezone?: string;
   canonicalUrl?: string;
+  unocss?: any // "object"
```
Comment (ije) [inline types.d.ts:33]: nice, can you please import the `UserConfig` types?

Comment (omar2205) [reply types.d.ts:33]: Sorry, not sure what you mean

Comment (ije) [reply types.d.ts]: i mean you can instead the `unocss?: any ` with `unocss?:UserConfig`, and the `UserConfig` can be imported by `import type { UserConfig as UnoConfig } from "https://esm.sh/@unocss/core@0.43.2";`

**Review by ije — APPROVED** [review]
LGTM

**Pull request merged**